MQ2AutoSize.cpp

```diff
@@ -1,110 +1,224 @@
-//  /autosize sizeself #	- Sets size for your character
-//
-//  /autosize status		 - Display current plugin settings to chatwnd
-//  /autosize help			- Display command syntax to chatwnd
-//  /autosize save			- Save settings to INI file (auto on plugin unload)
-//  /autosize load			- Load settings from INI file (auto on plugin load)
-//  /autosize autosave	  - Automatically save settings to INI file when an option is toggled or size is set
-//
-//////////////////////////////////////////////////////////////////////////////
+// MQ2AutoSize.cpp : Resize spawns by distance (client only)
 
 #include <mq/Plugin.h>
 
-const char* MODULE_NAME = "MQ2AutoSize";
-PreSetup(MODULE_NAME);
-PLUGIN_VERSION(1.0);
-// this controls how many pulses to perform a radius-based resize (bad performance hit)
-const int	SKIP_PULSES = 5;
-// min and max size values
-const float MIN_SIZE = 1.0f;
-const float MAX_SIZE = 250.0f;
-
-// used by the plugin
-const float OTHER_SIZE = 1.0f;
-const float ZERO_SIZE = 0.0f;
-unsigned int uiSkipPulse = 0;
-char szTemp[MAX_STRING] = { 0 };
-void SpawnListResize(bool bReset);
+// Constants
+constexpr const int SKIP_PULSES = 5;               // Controls the number of pulses to perform a radius-based resize
+constexpr const int MIN_SIZE = 1;                  // Minimum size value
+constexpr const int MAX_SIZE = 250;                // Maximum size value
+constexpr int FAR_CLIP_PLANE = 1000;               // Placeholder for EQ Far Clip Plane value
+constexpr const float OTHER_SIZE = 1.0f;           // Default size for other entities
+constexpr const float ZERO_SIZE = 0.0f;            // Size representing zero
+
+// Variables
+unsigned int uiSkipPulse = 0;            // Skip pulse counter
+char szTemp[MAX_STRING] = { 0 };         // Temporary buffer for strings
+int previousRangeDistance = 0;           // Previous range distance
+bool loaded_dannet = false;              // State of DanNet plugin
+bool loaded_eqbc = false;                // State of EQBC plugin
+uint64_t commsCheck =0;                  // Comms check timestamp
+
+// Function Declarations
+void SpawnListResize(bool bReset);       // Function to resize the spawn list
+void ChooseInstructionPlugin();          // Function to choose the instruction plugin
+void emulate(const std::string_view type);   // Function to emulate certain behavior (zonewide vs range)
+void DrawAutoSize_MQSettingsPanel();     // Function to draw the MQ settings panel
+void SendGroupCommand(const std::string_view who); // Function to send a command to a group
+int RoundToNearestTen(int value);        // Function to round a value to the nearest ten
+static bool isInGroup();                 // Function to check if in a group
+static bool isInRaid();                  // Function to check if in a raid
+
+// Plugin Setup
+PreSetup("MQ2AutoSize");
+PLUGIN_VERSION(1.1);
+
+enum class eCommunicationMode
+{
+	None = 0,
+	DanNet = 1,
+	EQBC = 2,
+
+	Default = 0
```

Comment:
You can write `Default = None` here for clarity. Same idea below with eResizeMode: `Default = Range`